lib/generators/rspec/scaffold/templates/request_spec.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal:true
+
+require 'rails_helper'
+
+# This spec was generated by rspec-rails when you ran the scaffold generator.
+# It demonstrates how one might use RSpec to specify the controller code that
+# was generated by Rails when you ran the scaffold generator.
+#
+# It assumes that the implementation code is generated by the rails scaffold
+# generator. If you are using any extension libraries to generate different
+# controller code, this generated spec may or may not pass.
+#
+# It only uses APIs available in rails and/or rspec-rails. There are a number
+# of tools you can use to make these specs even more expressive, but we're
+# sticking to rails and rspec-rails APIs to keep things simple and stable.
+
+<% module_namespacing do -%>
+RSpec.describe <%= controller_class_name %>Controller, <%= type_metatag(:request) %> do
+  # This should return the minimal set of attributes required to create a valid
+  # <%= class_name %>. As you add validations to <%= class_name %>, be sure to
+  # adjust the attributes here as well.
+  let(:valid_attributes) {
+    skip("Add a hash of attributes valid for your model")
+  }
+
+  let(:invalid_attributes) {
+    skip("Add a hash of attributes invalid for your model")
+  }
+
+<% unless options[:singleton] -%>
+  describe "GET /index" do
+    it "renders a successful response" do
+      <%= class_name %>.create! valid_attributes
+      get <%= "#{index_helper}_path" %>
+      expect(response).to be_successful
+    end
+  end
+<% end -%>
+
+  describe "GET /new" do
+    it "renders a successful response" do
+      get <%= "#{new_helper}_path" %>
+      expect(response).to be_successful
+    end
+  end
+
+  describe "POST /create" do
+    context "with valid parameters" do
+      it "creates a new <%= class_name %>" do
+        expect {
+          post <%= "#{index_helper}_path" %>, params: { <%= ns_file_name %>: valid_attributes }
+        }.to change(<%= class_name %>, :count).by(1)
+      end
+
+      it "redirects to the created <%= ns_file_name %>" do
+        expect {
+          post <%= "#{index_helper}_path" %>, params: { <%= ns_file_name %>: valid_attributes }
+        }.to redirect_to("<%= ns_table_name %>/#{<%= class_name %>.last}")
```

Comment:
Not so sure which style should be the convention. Should the specs use the paths for the URLs (at this point routing specs become a bit pointless) or use the rails helpers?

Reply:
I got guided by `rails/generators/test_unit/scaffold/templates/functional_test.rb.tt`